cpp/src/transform/encode.cpp

```diff
@@ -0,0 +1,56 @@
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/detail/search.hpp>
+#include <cudf/detail/stream_compaction.hpp>
+#include <cudf/detail/transform.hpp>
+#include <cudf/table/table_view.hpp>
+
+namespace cudf {
+namespace detail {
+
+std::pair<std::unique_ptr<column>, std::unique_ptr<column>> encode(
+  column_view const& input_column, rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+{
+  // side effects of this function were are now dependent on:
+  // - resulting column elements are sorted ascending
+  // - nulls are sorted to the beginning
+  auto table_keys = cudf::detail::drop_duplicates(table_view{{input_column}},
+                                                  std::vector<size_type>{0},
+                                                  duplicate_keep_option::KEEP_FIRST,
+                                                  null_equality::EQUAL,
+                                                  mr,
+                                                  stream)
+                      ->release();  // true == nulls are equal
```

Comment:
```suggestion
                      ->release();
```
This was caused by a reformatting all the code using clang-format some time ago